airgun/views/host.py

```diff
@@ -39,26 +40,68 @@ class TableActions(View):
 class PuppetClassParameterValue(Widget):
     """Represent value field for Puppet Class parameters table row from Host
     Parameters tab. That field can be interacted with as usual text input, but
-    also it can be overriden with new value
+    also it can be overridden with new value
     """
     ROOT = ".//div[@class='input-group']"
     value = TextInput(
-        locator=".//textarea[contains(@name, 'lookup_values_attributes')]")
-    override_parameter = Text(".//a[@data-tag='override']")
-    remove_override = Text(".//a[@data-tag='remove']")
+        locator=".//*[self::textarea or self::input][contains(@name, 'lookup_values_attributes')]")
+    override_button = Text(".//a[@data-tag='override']")
+    remove_override_button = Text(".//a[@data-tag='remove']")
+    hide_button = Text(".//a[contains(@class, 'btn-hide')]")
+
+    def _button_is_displayed(self, button_widget):
+        """Return whether the widget is displayed"""
+        return self.browser.wait_for_element(
+            button_widget, visible=True, exception=False, parent=self) is not None
+
+    @property
+    def hidden(self):
+        """Return whether the variable is hidden"""
+        return 'masked-input' in self.browser.classes(self.value)
+
+    @property
+    def overridden(self):
+        """Return whether the variable is overridden, a variable is overridden if not disabled"""
+        return not ('disabled' in self.browser.classes(self.value) or
+                    self.browser.get_attribute('disabled', self.value) == 'disabled' or
+                    self.browser.get_attribute('disabled', self.value) == 'true')
+
+    @property
+    def hidden_value(self):
+        return self.browser.get_attribute('data-hidden-value', self.value)
 
     def read(self):
-        return self.value.read()
+        """Return smart variable widget values"""
+        return {'value': self.value.read(), 'overridden': self.overridden, 'hidden': self.hidden,
```

Comment:
I think that we should only value or hidden value if it is hidden, but only value nevertheless

Reply:
1- There not only a text input in this Value, there is also many buttons, and we need that different properties in the tests.
2- this read format is compatible with fill , to override or hide a value.

Reply:
+1 to @ldjebran , `2)` is a good point.

Reply:
I am not so happy about that implementation, but if @abalakh fine - I am also fine =)
I feel that we can transform all that logic in the combination of different widgets and have proper decomposition, but we do not have perfection as a goal